### DISCUSSION TITLE
Record video with Collector

### DISCUSSION THREAD

**Discussion by ikamensh** [thu-ml/tianshou#762] [Q&A]
I want to record a video with a demo of my agent. Is there a recommended way to do so? I found that I can pass render flag to Collector.collect, but this is not enough.

```python
test_collector.collect(
        n_episode=test_num, render=True
    )
```

**Comment by Trinkle23897** [comment]
render=1/35 means recording video with 35 FPS, see the example scripts or docs 
https://tianshou.readthedocs.io/en/master/api/tianshou.data.html#tianshou.data.Collector.collect

**Comment by grahamseamans** [comment]
I want to log video to wandb. Does anyone have any suggestions on how to implement this?

something like this works for anyone interested. it ends up not being very pretty on wandb.com but I'm sure that can be fixed up as well.
```
test_envs = SubprocVectorEnv(
            [
                lambda: gym.wrappers.RecordVideo(
                    gym.make(args.task, render_mode="rgb_array"),
                    video_folder=f"videos/{logger.wandb_run.id}",
                    episode_trigger=lambda x: x % args.test_num == 0,
                )
            ]
       )
```
it saves one episode to video per testing period. With this solution you only have one testing env (not running testing envs in parallel)

**Comment by grahamseamans** [comment]
@ikamensh here's a way to do what you're trying to do. It will take some more dependencies but nothing crazy.

```

from tianshou.env import SubprocVectorEnv
def make_video():
    policy.eval()
    rec_env = SubprocVectorEnv(
        [
            lambda: gym.wrappers.RecordVideo(
                gym.make(args.task, render_mode="rgb_array"), video_folder="video/"
            )
        ]
    )
    rec_env.seed(args.seed)
    collector = Collector(policy, rec_env, exploration_noise=True)
    collector.collect(n_episode=1)
```
Change n_episode for more videos... Cheers 👍
